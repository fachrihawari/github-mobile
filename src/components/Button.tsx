import React, { ReactNode } from "react";
import {
  Text,
  StyleProp,
  TextStyle,
  StyleSheet,
  ViewStyle,
  View
} from "react-native";

import Touchable from "./Touchable";
import { color } from "../config";

interface IButtonProps {
  onPress: Function;
  wrapperStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  children: String;
}

const style = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: color.primary,
    paddingHorizontal: 20,
    paddingVertical: 10,
    borderRadius: 8
  },
  text: {
    color: "white",
    fontSize: 16
  }
});

function Button(props: IButtonProps) {
  return (
    <Touchable onPress={props.onPress}>
      <View style={[style.button, props.wrapperStyle]}>
        <Text style={[style.text, props.textStyle]}>{props.children}</Text>
      </View>
    </Touchable>
  );
}

export default Button;
