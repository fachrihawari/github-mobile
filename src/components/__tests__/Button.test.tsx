import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Button from '../Button';


describe('rendering', () => {
  
  let wrapper: ShallowWrapper
  const handlePress = jest.fn()
  const buttonLabel = 'Awesome Button'

  beforeEach(() => {
    wrapper = shallow(<Button onPress={handlePress}>{buttonLabel}</Button>)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should render <Touchable/>', () => {
    expect(wrapper.find('Touchable')).toHaveLength(1);
  });

  it('has correct button label', () => {
    expect(wrapper.find('Text').contains(buttonLabel)).toBe(true)
  })

  it('has pressed the button', () => {
    wrapper.simulate('press')
    expect(handlePress).toHaveBeenCalledTimes(1)
  })
})
