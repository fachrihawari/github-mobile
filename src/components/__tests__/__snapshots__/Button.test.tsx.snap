// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render correctly 1`] = `
<Touchable
  onPress={[MockFunction]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#0087E9",
          "borderRadius": 8,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Awesome Button
    </Text>
  </View>
</Touchable>
`;
